### PR TITLE
[LUA]Clears enmity when using Staging Point doors.

### PR DESCRIPTION
**Pull request by hooksta4**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?
- Changed startEvent() to startOptionalEvent() which will clear enmity.

## Steps to test these changes
Mob should not follow you into staging point. 
Go to staging point, aggro a mob, reenter staging point via door. Enmity will be lost for player and pet as per retail (No mobs in the champagne room!)

### CODE REVIEW

**Comment by Xaver-DaRed** [comment]
This functionality already exists
startCutscene()
startOptionalCutscene()

Both do that

**Comment by hooksta4** [comment]
fair enough.

**Comment by hooksta4** [comment]
There might be some other instances where we would want to clearEnmity from a function at some point maybe?

File: src/map/lua/lua_baseentity.h [hunk 1]
```diff
@@ -631,6 +631,7 @@ class CLuaBaseEntity
     void  setClaimable(bool claimable);
     bool  getClaimable();
     void  clearEnmityForEntity(CLuaBaseEntity* PEntity);
+    void  clearEnmity(CLuaBaseEntity* PEntity); // clears player enmity from notoriety list
```
Comment (WinterSolstice8) [inline src/map/lua/lua_baseentity.h]: This is actually clearing `EnmityContainer`, I assume this was an old comment (they have similar names but different functions)

**Comment by Xaver-DaRed** [comment]
Definately. Hence why I dont opose to implementing that function.
But I dont think we need to use it in those concrete places (having an actual event)

In situations where we dont have an event to trigger? definately

File: src/map/lua/lua_baseentity.h (hunk 1 above)
Comment (hooksta4) [reply src/map/lua/lua_baseentity.h]: fixed.

**Comment by hooksta4** [comment]
> Definately. Hence why I dont opose to implementing that function. But I dont think we need to use it in those concrete places (having an actual event)
> 
> In situations where we dont have an event to trigger? definately

Updated the SP with the startCutscene but left the clearEnmity() in.

File: scripts/zones/Arrapago_Reef/npcs/_jib.lua [hunk 2]
```diff
@@ -15,7 +15,7 @@ entity.onTrigger = function(player, npc)
         if player:getXPos() < 8 then
             player:messageSpecial(ID.text.STAGING_GATE_ILRUSI)
             player:messageSpecial(ID.text.STAGING_GATE_INTERACT)
-            player:startEvent(106)
+            player:startCutscene(106)
```
Comment (Xaver-DaRed) [inline scripts/zones/Arrapago_Reef/npcs/_jib.lua]: Ok, so, when an event returns an option ("wanna open this door?" "wanna use the portal?" etc) the function to use is `:startOptionalCutscene()`

The stagging point does ask you.

File: src/map/lua/lua_baseentity.cpp [hunk 3]
```diff
@@ -12362,6 +12362,31 @@ sol::table CLuaBaseEntity::getNotorietyList()
...
 
+/************************************************************************
+ *  Function: clearEnmity()
+ *  Purpose : Clears all enmity towards target
+ *  Example : mob:clearEnmity(target)
+ *  Notes   :
+ ************************************************************************/
+void CLuaBaseEntity::clearEnmity(CLuaBaseEntity* PEntity)
+{
+    (m_PBaseEntity->objtype != TYPE_MOB);
```
Comment (Xaver-DaRed) [inline src/map/lua/lua_baseentity.cpp]: ?

Was this a check to return at some point?

**Comment by hooksta4** [comment]
I am not sure why the CI is blowing me up? I don't get any errors on load.

File: src/map/lua/lua_baseentity.cpp (hunk 3 above)
Comment (Xaver-DaRed) [reply src/map/lua/lua_baseentity.cpp]: From ASB
imagen

Comment (hooksta4) [reply src/map/lua/lua_baseentity.cpp]: XI_DEBUG_BREAK oh that whole line should be gone then eh

Comment (Xaver-DaRed) [reply src/map/lua/lua_baseentity.cpp]: Well, you probably want to check for entity type, insead of deleting the check entirely

**Review by Xaver-DaRed — APPROVED** [review]
Repurposed to make stagging points doors clear enmity

**Pull request merged**